modules/sublimeBidAdapter.js

```diff
@@ -1,135 +1,282 @@
+
+  let commonPayload = {
+    sublimeVersion: SUBLIME_VERSION,
+    // Current Prebid params
+    prebidVersion: '$prebid.version$',
+    currencyCode: config.getConfig('currency.adServerCurrency') || DEFAULT_CURRENCY,
+    timeout: typeof bidderRequest !== 'undefined' ? bidderRequest.timeout : config.getConfig('bidderTimeout'),
+  };
+
+  // RefererInfo
+  if (bidderRequest && bidderRequest.refererInfo) {
+    commonPayload.referer = bidderRequest.refererInfo.referer;
+    commonPayload.numIframes = bidderRequest.refererInfo.numIframes;
+  }
+  // GDPR handling
+  if (bidderRequest && bidderRequest.gdprConsent) {
+    commonPayload.gdprConsent = bidderRequest.gdprConsent.consentString;
+    commonPayload.gdpr = bidderRequest.gdprConsent.gdprApplies; // we're handling the undefined case server side
+
+    // Injecting gdpr consent into sublime tag
+    window.sublime.gdpr = window.sublime.gdpr || {};
+    window.sublime.gdpr.injected = {
+      consentString: bidderRequest.gdprConsent.consentString,
+      gdprApplies: bidderRequest.gdprConsent.gdprApplies
+    };
+  }
+
+  return validBidRequests.map(bid => {
+    let bidHost = bid.params.bidHost || DEFAULT_BID_HOST;
+    let callbackName = (bid.params.callbackName || DEFAULT_CALLBACK_NAME) + '_' + bid.params.zoneId;
+    let protocol = bid.params.protocol || DEFAULT_PROTOCOL;
+    let sacHost = bid.params.sacHost || DEFAULT_SAC_HOST;
+
+    setState({
+      transactionId: bid.transactionId,
+      zoneId: bid.params.zoneId,
+      debug: bid.params.debug || false,
     });
-  },
-
-  /**
-     * Unpack the response from the server into a list of bids.
-     *
-     * @param {*} serverResponse A successful response from the server.
-     * @param {*} bidRequest An object with bid request informations
-     * @return {Bid[]} An array of bids which were nested inside the server.
-     */
-  interpretResponse: (serverResponse, bidRequest) => {
-    const bidResponses = [];
-    const response = serverResponse.body;
-
-    if (response) {
-      if (response.timeout || !response.ad || response.ad.match(/<!-- No ad -->/gmi)) {
-        return bidResponses;
-      }
 
-      // Setting our returned sizes object to default values
-      let returnedSizes = {
-        width: 1800,
-        height: 1000
-      };
+    // Adding Sublime tag
+    let script = document.createElement('script');
+    script.type = 'application/javascript';
+    script.src = 'https://' + sacHost + '/sublime/' + bid.params.zoneId + '/prebid?callback=' + callbackName;
+    document.body.appendChild(script);
 
-      // Verifying Banner sizes
-      if (bidRequest && bidRequest.data && bidRequest.data.w === 1 && bidRequest.data.h === 1) {
-        // If banner sizes are 1x1 we set our default size object to 1x1
-        returnedSizes = {
-          width: 1,
-          height: 1
-        };
+    // Register a callback to send notify
+    window[callbackName] = (response) => {
+      function querify(params) {
```

Comment:
This function declaration could/should be in the top scope, juste like other utils.

Moreover, I think this is redundant with url.formatQS, which is already imported.

Reply:
I removed this function as it used only one time